### PR TITLE
#184637234 Configure The App to use PostgreSQL as DB Engine and Sequelize ORM

### PR DESCRIPTION
**Pull request by masterchief-00**
#### What does this PR do?

- Configures the project to use postgrSQL as the database and Sequelize ORM

#### Description of Task to be completed?
- Configure The App to use PostgreSQL as DB Engine and Sequelize ORM
#### How should this be manually tested?
 - clone the project and use the terminal to start the development server by "npm run dev"
 - make sure that you have PostgreSQL installed
 - create a database and
 - then use the database name and password to make the DEV_DATABASE_URL in your env file,
 - refer to the env.example file
#### Any background context you want to provide?
 none
#### What are the relevant pivotal tracker/Trello stories?
 Tracker: 184637234
#### Screenshots (if appropriate)
 none
#### Questions:
 none

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "presets": [
+      ["env", {
+        "targets": {
+          "node": "current"
+        }
+      }]
+    ]
+  }
```
Comment (fab-ryan) [inline .babelrc]: please remember to put new line at the end

File: src/config/database.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import * as dotenv from "dotenv";
+dotenv.config();
+import Sequelize from "sequelize";
+
+export const sequelize = new Sequelize(process.env.DEV_DATABASE_URL, {
```
Comment (fab-ryan) [inline src/config/database.js]: you have config file then again you are calling env, use config file to determine node environment

File: src/models/index.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+const config = configFile[env];
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+export const db = {};
+
+let sequelize;
+
+if (config.use_env_variable) {
+  sequelize = new Sequelize(process.env[config.use_env_variable], config);
```
Comment (fab-ryan) [inline src/models/index.js]: referencing to this

File: server.js [hunk 4]
```diff
@@ -2,12 +2,14 @@ import * as dotenv from "dotenv";
...
 import welcomeRoutes from "./src/routes/welcomeRoute.js";
+import dummyRoutes from "./src/routes/dummyRoutes.js";
 
 export const app = express();
 
 app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
 app.use("/welcome", welcomeRoutes);
+app.use("/users", dummyRoutes);
```
Comment (fab-ryan) [inline server.js]: you can combine all routes and calling it here once

**Review by fab-ryan — CHANGES_REQUESTED** [review]
work to those comments

File: index.js [hunk 5]
```diff
@@ -1,7 +1,24 @@
...
 dotenv.config();
 import app from "./server.js";
+<<<<<<< HEAD
+=======
+import db from "./src/models/index.js";
+>>>>>>> c954de6 (parent 67e0d70c79f35a919326b1bc05c4d4986527c9c7)
 
 app.listen(process.env.PORT, () => {
   console.log(`-->Port ${process.env.PORT}: the server is up and running!`);
 });
```
Comment (fab-ryan) [inline index.js]: what???

File: src/config/database.js (hunk 2 above)
Comment (fab-ryan) [inline src/config/database.js]: you didn't change it?

**Review by fab-ryan — APPROVED** [review]

**Review by kirengaboris — APPROVED** [review]

**Comment by istafrica-dennw** [comment]
@masterchief-00 , also set up the babel for es6

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]
1. Add babel
2. your commit?? named how?? rename it t follow conventions

File: src/middlewares/middleware.js [hunk 6]
```diff
@@ -1,4 +1,8 @@
 /**
  * eg: MIDDLEWARE 1
  * not a real file
- */
\ No newline at end of file
+<<<<<<< HEAD
+ */
+=======
+ */
```
Comment (fab-ryan) [inline src/middlewares/middleware.js]: remove those unnecessary line

File: src/models/User.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+      },
+    },
+    token: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+  });
+
+  return User;
+};
```
Comment (fab-ryan) [inline src/models/User.js]: remember this at every time

File: src/controllers/dummyController.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import db from "../models/index.js";
```
Comment (fab-ryan) [inline src/controllers/dummyController.js]: it seems you haven't configured  babel well

**Review by fab-ryan — APPROVED** [review]

**Comment by masterchief-00** [comment]
> 1. Add babel
> 2. your commit?? named how?? rename it t follow conventions

It's all done

**Review by istafrica-dennw — APPROVED** [review]

**Pull request merged**